hydra_agent/classes_objects.py

```diff
@@ -188,6 +188,13 @@ def load_from_server(
         # save the new data.
         self.redis_graph.commit()
 
+    def connect_nodes(self, endpoint_properties, node_alias):
+        for class_endpoint in endpoint_properties:
+            src_node = node_alias.get(class_endpoint)
+            for endpoint in endpoint_properties[class_endpoint]:
+                node = node_alias.get(endpoint)
```

Comment:
You are considering that `node_alias` have all nodes required for an edge. But for "classes property endpoint", the node can be anywhere in the graph. It is not necessary that it will be in class endpoints`node_alias`. So, We need to traverse the whole graph to find that property node.

Reply:
@sandeepsajan0 all the nodes for class_endpoints are being created by a single call to `endpointclasses` method. As soon as each node is created, it's appended `node_alias`. The above implementation is only for adding edge between `entrypoint` node and `classEndpoint` (that's the only function that `endpointclasses` method serves). `object_property` method hasn't been changed

Reply:
line 232 is responsible for `Entrypoint` and `classEndpoint` connection. But the code you have changed is responsible for connecting the "classEndpoints nodes" to the "object_property nodes" which may be available already. See, `State` is an `object_property` for any "collection member" and also for "classEndpoint". So, It can happen that there will be a node of `object_property`. So, We don't need to create duplicate node else we have to search for that node and connect to that.

Reply:
@sandeepsajan0 I think that the code I changed connected `classEndpoint`s to other `classEndpoint`s. If I have 2 `classEndpoint`s and one of them has a property whose value is equal to the other `classEndpoint`, the code should create an edge between them. Please correct me if I'm wrong. If this is so, then `node_alias` method should work

Reply:
Yes, @ShravanDoda  You are partially right. It may be 2 `classEndpoints` but it may also be a `property`. See the example, in hydrus/doc.json file there is a class `State` which is also a property of `Command` and other classes. But `State` can not get from `Entrypoint` endpoint.
So, It is not the same as to connect 2 classes node. And I guess there may still a small bug but not same as we are discussing. Try to read apidoc and then try to find or solve the bug. I will ping you if I'll get the bug.
This [documentation](https://medium.com/w3c-hydra-development-community/documentation-for-hydra-graph-cd9b2bd84884) may help you.

Reply:
@ShravanDoda As I can see, there is no bug because in `hydra_graph` module we first calling to `classes_enpoints` and then `collection_endpoints`. So, There cannot be a property which already present in the graph. So, we don't need to traverse the whole graph. I think, your code should work the same. I will check and merge it after merging #87.